test/ruby/test_iseq.rb

```diff
@@ -809,7 +809,6 @@ def test_ever_condition_loop
   def test_unreachable_syntax_error
     mesg = /Invalid break/
     assert_syntax_error("false and break", mesg)
-    assert_syntax_error("if false and break; end", mesg)
```

Comment:
Why removing this assertion?

Reply:
Since the next part of the conditional is no longer compiled, it won't trigger that syntax error anymore. For the line just above it, it enters a different code path since there is no `if` and I didn't write this short-circuit optimization in that part of the code.

Reply:
The `break` exists syntactically, and should be a subject of checks.

Reply:
If that's the case I'll try to come up with a different solution to the problem.